src/kernels/MIOpenArgmax.cpp

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef MIOPEN_DONT_USE_HIP_RUNTIME_HEADERS
+#include <hip/hip_fp16.h>
+#include <hip/hip_runtime.h>
+#endif
+
+#include "float_types.h"
+
+extern "C" __global__ void ArgmaxFwdContiguous(const FLOAT* __restrict__ x,
+                                               int32_t* __restrict__ y,
+                                               uint64_t output_numel,
+                                               int32_t reduce_size,
+                                               uint64_t inner_size,
+                                               int32_t dim)
+{
+    const uint64_t gid = threadIdx.x + blockIdx.x * blockDim.x;
+    if(gid >= output_numel)
```

Comment:
Kind of important this for huge tensors - since gridsize is limited by 4,294,967,295 it is potentially possible to hit this limit. Probably no right now, but still possible.
May I ask you to add additional loop to handle this situation (at least in the kernel, since it looks trivial)?

As about the host code - it's bit more complicated question how to limit grid size properly. I guess it has to be handled separately.

#2593 shows an example where we a close to that problem.

Reply:
To solver that problem. After calculating the gridsize in the host code, if it is larger than the maximum value, I need to calculate the size to loop inside the kernel and pass it to the kernel to use it.
I wonder if all future PRs should consider this?

Reply:
Ideally it should be considered for future kernels. But probably we need to make some clear recommendations and examples and update existing kernels as well.

Reply:
@seungmanhan 
> I wonder if all future PRs should consider this?

Yes. But in this solver -- please make it inapplicable if grid size is too big.

> After calculating the gridsize in the host code, if it is larger than the maximum value, I need to calculate the size to loop inside the kernel and pass it to the kernel to use it.

[future] I recommend considering performance. If that grid-size-limitation-free solution would have significantly worse performance, then I highly recommend creating a new solver (while keeping the old, performant one).

Reply:
I think this review comment is [resolved], but there is a new review comment related to this. @seungmanhan Thanks for quick responses!